utils/server/allocatedWorkers.js

```diff
@@ -20,19 +20,17 @@ export const getResidentAllocatedWorkers = async (mosaic_id, params) => {
   };
 };
 
-export const deleteResidentAllocatedWorker = async (id, body) => {
-  /* const { data } = await axios.patch(
-   `${ENDPOINT_API}/allocations/delete`,
-   formData,
-   {
-     headers: { 'Content-Type': 'application/json', 'x-api-key': AWS_KEY },
-   }
- );
- return { ref: data?.['_id'] }; */
-  await new Promise((resolve) =>
-    setTimeout(() => resolve(console.log('data is ' + id + body)), 3000)
-  );
-  return { test: 'test' };
+const deleteAllocatedWorkerSchema = yup.object().shape({
+  id: yup.number().required().integer(),
+  deallocationReason: yup.string().required(),
+});
+
+export const deleteAllocatedWorker = async (params) => {
+  const body = await deleteAllocatedWorkerSchema.validate({ ...params });
```

Comment:
```suggestion
  const body = await deleteAllocatedWorkerSchema.validate(params);
```

Spreading a recreating an object is an overhead here.

Reply:
👍